lock-impl/src/main/java/com/palantir/lock/impl/LockServiceImpl.java

```diff
@@ -1009,6 +1005,32 @@ public void logCurrentState() {
     }
 
     @Override
+    public void logCurrentHeldLocks(String descriptorRegex, boolean versionIdNecessary) {
+        StringBuilder logString = getGeneralLockStats();
```

Comment:
This appears to be where the meat of the changes lives. The class is already pretty huge. It would be good to extract a little class whose sole responsibility is to generate this `logString` (as a String, not just a builder).

Then this method can become something like:
`String logString = new LockStatsGenerator(...).getStats(...);
log.error(...);`

And testing it will be significantly easier (and you won't have to touch the horrible `LockServiceImplTest` :)